### DISCUSSION TITLE
Read Library Error v1.8.0

### DISCUSSION THREAD

**Discussion by Leyakk** [AntonelloDN/Morpho#19] [Bugs]
Hello,

I'm having the same error with the v1.8.0 and v1.7.0 of Morpho ... even when i open the example files 
For the Morpho library component it's saying "1. Solution exception:Le caractère '�', valeur hexadécimale 0xFFFD, ne peut pas être inclus dans un nom. Ligne 798, position 16."
and for the Morpho buidling "1. Solution exception:expected Grid, got FaceGroup"

Does anyone else have the same problem? how can I fix it ?

Thank you

**Comment by AntonelloDN** [comment]
Hi @Leyakk, the first error message seems to be related to Envimet DB. Can you tell me your current Envimet version?
The second error message should be a conflict of Morpho version. But let's start with the first one

**Reply by Leyakk** [reply]
Thank you for your reply !
I'm using the Envimet 5.5.1

**Reply by AntonelloDN** [reply]
Hi @Leyakk, 
I did a test with a fresh installation of both Envimet and Morpho and I can't replicate the issue but I think I know where it comes from. There is an latin char in the custom XML of Envimet `database.edb` that isn't marked with the proper character encoding scheme.
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/6e139af8-bd01-495e-8952-6782f99c74cf)

Morpho should handle it correctly but for some reason it does not happen in your case.
can you confirm that you only have Morpho.Rhino 2.0.0 and Envimet 5.5.1 on your machine?

If you have old versions of Morpho please remove them.

From Add/Remove programs
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/b19e3356-74d3-4821-8c00-698e036736b0)
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/2b7c4598-7601-4bc2-baff-bf62516895d1)


If you go to `%APPDATA%\Morpho` you should see something like this
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/b75317c0-2343-4b53-a7f2-8f058f921ece)
If you check the property of one of the DLL you should confirm the version 2.0.0
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/e20c0a8c-f25c-4bcc-a855-c29f6df5ee3d)

**Reply by Leyakk** [reply]
Hello ! Thank you ! it worked when i deleted the older versions of Envi-met on my PC

**Comment by Leyakk** [comment]
@AntonelloDN Actually, it's not working anymore!
I have installed the new Envi-met and deleted all the rest and same for Morpho ... it was working at the beginning until I've put a new userdatabase in Envimet, the library component would start giving me this error 
1. Solution exception:Le caractère '�', valeur hexadécimale 0xFFFD, ne peut pas être inclus dans un nom. Ligne 798, position 16.

**Reply by AntonelloDN** [reply]
Thanks @Leyakk, can you help me to replicate the issue?
Is it a specific user DB that you are importing?
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/205449fd-677e-4812-8ec4-ddba34cc86e6)

**Reply by Leyakk** [reply]
It's a userdatabase that I modified a bit  (not very complicated ) https://we.tl/t-WZDzV8Mu9I
I also tried to remove the "é" and spaces from the material names but it still not working when I import it.

As a temporary solution, I went back to a previous version of Envi-met (V5.1.2) and an older version of morpho (1.0.1) and they are working fine

**Reply by AntonelloDN** [reply]
Thanks! @Leyakk Can you try the new installer https://github.com/AntonelloDN/Morpho/releases/tag/v1.9.0 ?

**Reply by AntonelloDN** [reply]
![image](https://github.com/AntonelloDN/Morpho/assets/16763953/5264dfe7-dbc1-4021-b1b2-6cb85b9cf08b)

**Reply by Leyakk** [reply]
Thank you very much !! I have a very long calculation running with the older version right now ! I will try and let you know as soon as I can !

**Reply by nvskumar-m** [reply]
![image](https://github.com/AntonelloDN/Morpho/assets/54622340/8a7f4d94-f6f9-4f75-8426-224b59444191)

I see the same issue in my system. My license is under renewal process; probably lite version may be the reason for error?

**Comment by nvskumar-m** [comment]
![image](https://github.com/AntonelloDN/Morpho/assets/54622340/4c86bd4a-12d1-4e4e-a635-66622b1aaa17)
 
I see same issue again even with new installer. As my license is under renewal process, I am using lite version. Could it be the reason for error?

**Comment by KeviChiu** [comment]
Hey Antonello, 
My version of Morpho is 1.9.0 and ENVI-MET is student license of 5.5.1. The error is the same with [nvskumar-m] that shown on here. 
Hopping any solution or new version for that.

Thanks,  
Kev.

**Reply by AntonelloDN** [reply]
Hi @nvskumar-m, @KeviChiu,
Can you test this new [installer](https://github.com/AntonelloDN/Morpho/releases/tag/v1.10.0)? Thank you

**Reply by nvskumar-m** [reply]
Hi Antonello, Sorry for the late response. But the system is warning of the installation of a virus the moment I download your recommended version.

![image](https://github.com/AntonelloDN/Morpho/assets/54622340/e57f1807-07c4-4038-8e6a-0a0e090dd691)

**Reply by AntonelloDN** [reply]
Hi @nvskumar-m, It Is a false positive. I use a free software to generate the installer and the binary Is not signed - the sign service is not for free, usually it is used for commercial products. This is the reason of the warnings

The quick solution in this case is:

- skip the warnings and install it anyway

In the meantime, I can search for the cheaper provider for signing the code. But we can continue this discussion in a separate thread. It is not a bug.

**Reply by nvskumar-m** [reply]
Thank you, and sorry for the confusion.
